docker/docker_image_dest_test.go

```diff
@@ -6,10 +6,13 @@ import (
 	"net/http"
 	"testing"
 
+	internalTypes "github.com/containers/image/v5/internal/types"
 	"github.com/stretchr/testify/assert"
 	"github.com/stretchr/testify/require"
 )
 
+var _ internalTypes.ImageDestinationWithOptions = &dockerImageDestination{}
```

Comment:
Because of the type checks, we need to have some mechanisms ensure the conformances hold, but it would be nice to be able to do that without relying on manually writing tests like this. (I did break `ImageDestinationPartial` in one version of this PR already.)

Hopefully a new internal `Transport` (with `ParseInternalReference` returning an internal `Reference`, and `NewInternalSource` returning an internal `ImageSource`, will do that automatically, or something like that.